nifi-commons/nifi-security-utils/src/main/java/org/apache/nifi/security/util/KeyDerivationFunction.java

```diff
@@ -25,8 +25,11 @@
 public enum KeyDerivationFunction {
 
     NIFI_LEGACY("NiFi legacy KDF", "MD5 @ 1000 iterations"),
-    OPENSSL_EVP_BYTES_TO_KEY("OpenSSL EVP_BytesToKey", "Single iteration MD5 compatible with PKCS#5 v1.5");
-    // TODO: Implement bcrypt, scrypt, and PBKDF2
+    OPENSSL_EVP_BYTES_TO_KEY("OpenSSL EVP_BytesToKey", "Single iteration MD5 compatible with PKCS#5 v1.5"),
+    BCRYPT("Bcrypt", "Bcrypt with configurable work factor: see https://cwiki.apache.org/confluence/display/NIFI/Key+Derivation+Function+Explanations"),
```

Comment:
The resources listed on the wiki are super helpful. I think it would make sense for these to additionally get rolled into the administrator's guide so we can have a one stop shop for all of it.